packages/core/src/modules/employment/queries/list-job-offers.ts

```diff
@@ -0,0 +1,73 @@
+import { ListSearchParams } from '@/shared/types';
+import { Company, EmploymentType, JobOfferStatus } from '../employment.types';
+import { db } from '@/infrastructure/database';
+import { sql } from 'kysely';
+import { z } from 'zod';
+import { ExtractValue } from '@oyster/types';
+
+const JobOfferFilterParams = z.object({
+  company: Company.shape.id.nullable().catch(null),
+  employmentType: z.nativeEnum(EmploymentType),
+  location: z.string().trim().min(1).nullable().catch(null),
+  status: z.nativeEnum(JobOfferStatus),
+});
+
+const JobOfferFilterParam = JobOfferFilterParams.keyof().enum;
+
+const JobOfferSearchParams = ListSearchParams.omit({
+  timezone: true,
+  search: true,
+}).merge(JobOfferFilterParams);
+
+type JobOfferFilterParam = ExtractValue<typeof JobOfferFilterParam>;
+type JobOfferSearchParams = z.infer<typeof JobOfferSearchParams>;
+
+async function listJobOffers({
+  company,
+  employmentType,
+  location,
+  status,
+}: JobOfferSearchParams) {
+  const query = db
+    .selectFrom('jobOffers')
+    .$if(!!company, (query) => {
+      return query.where((eb) => {
+        return eb.exists(
+          eb
+            .selectFrom('jobOffers')
+            .where('companies.crunchbaseId', '=', company)
+        );
+      });
+    })
```

Comment:
I think we can actually simplify this:

```suggestion
    .$if(!!company, (query) => {
      return query.where('jobOffers.companyId', '=', company)
    })
```